src/Microsoft.DotNet.Wpf/src/Shared/MS/Internal/IO/Packaging/PackagePartExtensions.cs

```diff
@@ -0,0 +1,85 @@
+        /// </summary>
+        /// <param name="packPart"></param>
+        /// <param name="mode">The FileMode to open the PackagePart</param>
+        /// <returns>A seekable stream representing the data in the PackagePart.</returns>
+        internal static Stream GetSeekableStream(this PackagePart packPart, FileMode mode)
+        {
+            return GetSeekableStream(packPart, mode, packPart.Package.FileOpenAccess);
+        }
+
+        /// <summary>
+        /// Gets a seekable stream from the PackagePart.
+        /// </summary>
+        /// <remarks>
+        /// In .NET Core 3.0, System.IO.Packaging was removed, in part, from WPF.  WPF now uses the implementation
+        /// contained in System.IO.Packaging.dll.  This implementation has distinct differences from the .NET Framework
+        /// WPF implementation.  One such difference is that the <see cref="DeflateStream"/> returned by <see cref="PackagePart.GetStream"/> calls
+        /// when the <see cref="PackagePart"/> is opened read-only is not a seekable stream.  This breaks several assumptions in WPF
+        /// and causes crashes when various parts of the code-base call into <see cref="Stream.Seek"/> or <see cref="Stream.Position"/>.
+        /// 
+        /// To fix this, we read the entire <see cref="DeflateStream"/> into a <see cref="MemoryStream"/>, allowing callers to fully seek the stream.
+        /// This is, generally, what would be the case in .NET Framework.
+        /// 
+        /// Note that if the stream returned is seekable (the <see cref="PackagePart"/> was opened write or read-write) then we just pass the resulting
+        /// stream back as we're already guaranteed it meets our needs.
+        /// </remarks>
+        /// <param name="packPart"></param>
+        /// <param name="mode">The FileMode to open the PackagePart</param>
+        /// <param name="access">The FileAccess used to open the PackagePart</param>
+        /// <returns>A seekable stream representing the data in the PackagePart.</returns>
+        internal static Stream GetSeekableStream(this PackagePart packPart, FileMode mode, FileAccess access)
+        {
+            var packStream = packPart.GetStream(mode, access);
+
+            // If the stream returned is seekable it meets all requirements and can be used directly.
+            if (packStream.CanSeek)
+            {
+                return packStream;
+            }
+
+            // Non-seekable streams need to be copied out into memory so they are seekable.
+            using (packStream)
+            {
+                var seekableStream = new MemoryStream((int)packStream.Length);
+
+                packStream.CopyTo(seekableStream);
+
+                // Reset the stream to the beginning.  If this is not done, attempts to read the stream 
+                // from the current position will fail.  E.G. XAML/XML parsing.
+                seekableStream.Position = 0;
```

Comment:
seekableStream.Seek(0, SeekOrigin.Begin)

Reply:
This is generally equivalent, is there a specific reason you prefer seek?  There are other places in the code where we use Position over a Seek call.

Reply:
Didn’t notice earlier we used Position elsewhere. Yes let’s use that for consistency.